pkg/operator/gcwatchercontroller/gcwatcher_controller_test.go

```diff
@@ -243,3 +254,102 @@ func TestCheckFiringAlerts(t *testing.T) {
+		operatorClient:         v1helpers.NewFakeStaticPodOperatorClient(&operatorv1.StaticPodOperatorSpec{OperatorSpec: operatorv1.OperatorSpec{ManagementState: operatorv1.Managed}}, status, nil, nil),
+		configMapClient:        nil,
+		alertNames:             []string{"dummy"},
+		alertingRulesCache:     []prometheusv1.AlertingRule{},
+		alertingRulesCacheLock: sync.RWMutex{},
+	}
+	prometheusResponseError := fmt.Errorf("error querying alerts: prometheus querying failed")
+	type test struct {
+		name                    string
+		clusterMonitoringExists bool
+		isPrometheusEnabled     bool
+		gc                      *GarbageCollectorWatcherController
+		expectErr               bool
+		expectedErrorMsg        error
+		expectedStatusCondition operatorv1.OperatorCondition
+	}
+	tests := []test{
+		{
+			name:                    "Garbage Collector Watcher with monitoring disabled",
+			clusterMonitoringExists: false,
+			gc:                      gcw,
+			expectErr:               false,
+			expectedStatusCondition: operatorv1.OperatorCondition{},
+		},
+		{
+			name:                    "Garbage Collector Watcher with monitoring enabled but with failing prometheus client",
+			clusterMonitoringExists: true,
+			isPrometheusEnabled:     false,
+			gc:                      gcw,
+			expectErr:               false,
+			expectedStatusCondition: successCondition,
+		},
+		{
+			name:                    "Garbage Collector Watcher with monitoring enabled but with correct prometheus client",
+			clusterMonitoringExists: true,
+			isPrometheusEnabled:     true,
+			gc:                      gcw,
+			expectErr:               true,
+			expectedErrorMsg:        prometheusResponseError,
+			expectedStatusCondition: failureCondition,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+			var monitoringName string
+			if tc.clusterMonitoringExists {
+				monitoringName = "monitoring"
+			}
+			if tc.isPrometheusEnabled {
+				promConnectivity := prometheusConnectivity{useCachedClient: true,
+					client: newFakePrometheusClient([]string{}, syncError)}
+				tc.gc.promConnectivity = promConnectivity
+			}
+			if err := indexer.Add(&configv1.ClusterOperator{
+				ObjectMeta: metav1.ObjectMeta{Name: monitoringName},
+			}); err != nil {
+				t.Fatal(err.Error())
+			}
+			listers := configlisters.NewClusterOperatorLister(indexer)
+			tc.gc.clusterLister = listers
+			eventRecorder := events.NewInMemoryRecorder("dummy")
+			syncContext := factory.NewSyncContext(controllerName, eventRecorder)
+			syncContext.Queue().Add(invalidateAlertingRulesCacheKey)
+			err := tc.gc.sync(context.TODO(), syncContext)
+			if tc.expectErr {
+				if err == nil {
+					t.Fatal("expected error but did not get one")
+				} else if tc.expectedErrorMsg.Error() != err.Error() {
+					t.Fatalf("expected err %v but got %v", tc.expectedErrorMsg, err)
+				}
+			}
```

Comment:
let's check if we got an error in the else branch

Reply:
I think this is what you wanted?

Reply:
+1